Adyen/Core/Payment Methods/PaymentMethod.swift

```diff
@@ -64,5 +64,7 @@ public func == (lhs: StoredPaymentMethod, rhs: StoredPaymentMethod) -> Bool {
 
 /// :nodoc:
 public func == (lhs: PaymentMethod, rhs: PaymentMethod) -> Bool {
-    lhs.type == rhs.type && lhs.name == rhs.name
+    lhs.type == rhs.type &&
+        lhs.name == rhs.name &&
+        String(describing: type(of: lhs)) == String(describing: type(of: rhs))
```

Comment:
that's actually correct, the indentation is signal that the second and third lines are actually extension of the first line. that's swift format, I didn't do it manually